From: Hilko Bengen <bengen@hilluzination.de>
Date: Tue, 11 Aug 2015 22:11:04 +0200
Subject: Fix autogenerated certificates

(1) curl (versions 7.38 and 7.43 as shipped with Debian stable and
unstable, respectively) won't establish TLS connections with a server
that presents a CN-less certificate.

(2) Since "Host" is expected to be an IP address, the CN should
reflect this address and IPAddresses should be set in addition to the
DNSNames field.
---
 certs/certs.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/certs/certs.go b/certs/certs.go
index 14511ee..69d0044 100644
--- a/certs/certs.go
+++ b/certs/certs.go
@@ -27,6 +27,7 @@ import (
 	"fmt"
 	"io/ioutil"
 	"math/big"
+	"net"
 	"os"
 	"time"
 )
@@ -57,6 +58,7 @@ func WriteNewCerts(certFile, keyFile string, server bool) error {
 		SerialNumber: serialNumber,
 		Subject: pkix.Name{
 			Organization: []string{"Stenographer"},
+			CommonName:   "127.0.0.1",
 		},
 		NotBefore: time.Now(),
 		NotAfter:  time.Now().Add(validFor),
@@ -64,6 +66,7 @@ func WriteNewCerts(certFile, keyFile string, server bool) error {
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 		BasicConstraintsValid: true,
 		DNSNames:              []string{"localhost"},
+		IPAddresses:           []net.IP{net.IPv4(127, 0, 0, 1)},
 		IsCA:                  true, // we're self-signed.
 	}
 	var keyFileMode os.FileMode
